File: scripts/retry_93.py
import json
import io
import urllib.request
from pypdf import PdfReader

# Load results
with open('/tmp/pagecount_results.json') as f:
    results = json.load(f)

# Get the failed ones
failed = [r for r in results if r['err']]
print(f'Failed: {len(failed)}')
for f in failed:
    print(f'  #{f["num"]}: {f["err"]}')

# Find the file keys for these
import subprocess
result = subprocess.run(
    ['node', '-e', f'''
require('dotenv').config({{ path: "/workspace/edutunisie/.env.local" }});
const {{ PrismaClient }} = require('/workspace/edutunisie/node_modules/@prisma/client');
const p = new PrismaClient({{ log: ['error'] }});
(async () => {{
  const files = await p.$queryRaw`SELECT r.id, r."numericId" as num, r."fileKey" as key FROM "Resource" r WHERE r."numericId" IN ({','.join(str(f['num']) for f in failed)})`;
  console.log(JSON.stringify(files));
  await p.$disconnect();
}})();
'''],
    capture_output=True, text=True
)
import re
out = result.stdout
idx = out.find('[')
if idx > 0:
    out = out[idx:]
files = json.loads(out)
print(f'Files: {len(files)}')

TOKEN = 'devmanet-bulk-2026'

for f in files:
    url = f'https://examanet.com/api/blob-teacher/{f["key"]}'
    try:
        req = urllib.request.Request(url, headers={'x-internal-token': TOKEN})
        with urllib.request.urlopen(req, timeout=30) as resp:
            data = resp.read()
        print(f'#{f["num"]}: {len(data)} bytes, first 4: {data[:4]!r}')
        if data[:4] == b'%PDF':
            reader = PdfReader(io.BytesIO(data))
            print(f'  Pages: {len(reader.pages)}')
    except Exception as e:
        print(f'#{f["num"]}: ERROR {e}')

File: scripts/retry_93.js
require('dotenv').config({ path: '/workspace/edutunisie/.env.local' });
const { PrismaClient } = require('/workspace/edutunisie/node_modules/@prisma/client');
const https = require('https');
const { PDFDocument } = require('/workspace/edutunisie/node_modules/pdf-lib');
const p = new PrismaClient({ log: ['error'] });
const TOKEN = process.env.INTERNAL_BULK_TOKEN || 'devmanet-bulk-2026';

function download(url, timeout=30000) {
  return new Promise((resolve, reject) => {
    const req = https.get(url, { headers: { 'x-internal-token': TOKEN }, timeout }, (res) => {
      if (res.statusCode === 301 || res.statusCode === 302) {
        return download(res.headers.location).then(resolve).catch(reject);
      }
      if (res.statusCode !== 200) return reject(new Error(`HTTP ${res.statusCode}`));
      const chunks = [];
      res.on('data', c => chunks.push(c));
      res.on('end', () => resolve(Buffer.concat(chunks)));
    });
    req.on('error', reject);
    req.on('timeout', () => { req.destroy(); reject(new Error('timeout')); });
  });
}

(async () => {
  for (const numId of [93, 2354, 4972, 4973]) {
    const r = await p.$queryRaw`SELECT r.id, r."numericId", r."fileKey" FROM "Resource" r WHERE r."numericId" = ${numId}`;
    if (!r[0]) continue;
    console.log(`Trying #${numId}...`);
    try {
      const url = `https://examanet.com/api/blob-teacher/${r[0].fileKey}`;
      const buf = await download(url);
      if (buf.slice(0, 4).toString() === '%PDF') {
        const pdf = await PDFDocument.load(buf, { ignoreEncryption: true });
        const newCount = pdf.getPageCount();
        await p.resourceContent.update({ where: { resourceId: r[0].id }, data: { pageCount: newCount } });
        console.log(`  Updated to ${newCount}`);
      } else {
        console.log(`  Not a PDF, skipping (first bytes: ${buf.slice(0, 4).toString()})`);
      }
    } catch (e) {
      console.log(`  ERROR: ${e.message}`);
    }
  }
  await p.$disconnect();
})();
